### ISSUE TITLE
Fix non-renderloop leaks in VulkanRenderer

### ISSUE THREAD

**Issue by skalarproduktraum** [ClearVolume/scenery#48]
By virtue of lwjgl's DebugAllocator, these leaks are

```
	at graphics.scenery.backends.vulkan.VU$VU.setImageLayout(VU.kt:206)
	at graphics.scenery.backends.vulkan.VU$VU.setImageLayout(VU.kt:264)
	at graphics.scenery.backends.vulkan.VulkanBuffer.<init>(VulkanBuffer.kt:16)
	at graphics.scenery.backends.vulkan.VulkanBuffer.<init>(VulkanBuffer.kt:9)
	at graphics.scenery.backends.vulkan.VulkanCommandBuffer.<init>(VulkanCommandBuffer.kt:13)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer$VulkanFramebufferAttachment.<init>(VulkanFramebuffer.kt:39)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer$VulkanFramebufferAttachment.<init>(VulkanFramebuffer.kt:40)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer$VulkanFramebufferAttachment.<init>(VulkanFramebuffer.kt:44)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.<init>(VulkanFramebuffer.kt:25)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.<init>(VulkanFramebuffer.kt:26)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.<init>(VulkanFramebuffer.kt:27)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.addSwapchainAttachment(VulkanFramebuffer.kt:276)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.createAttachment(VulkanFramebuffer.kt:50)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.createRenderpassAndFramebuffer(VulkanFramebuffer.kt:314)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.createRenderpassAndFramebuffer(VulkanFramebuffer.kt:323)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.createRenderpassAndFramebuffer(VulkanFramebuffer.kt:333)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.createRenderpassAndFramebuffer(VulkanFramebuffer.kt:343)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.createRenderpassAndFramebuffer(VulkanFramebuffer.kt:363)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.createRenderpassAndFramebuffer(VulkanFramebuffer.kt:377)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.getAttachmentDescBuffer(VulkanFramebuffer.kt:298)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.getAttachmentImageViews(VulkanFramebuffer.kt:305)
	at graphics.scenery.backends.vulkan.VulkanFramebuffer.getBestDepthFormat(VulkanFramebuffer.kt:441)
	at graphics.scenery.backends.vulkan.VulkanObjectState$texturesToDescriptorSet$1.accept(VulkanObjectState.kt:71)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:26)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:31)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:42)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:46)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:51)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:57)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:62)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:67)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:85)
	at graphics.scenery.backends.vulkan.VulkanPipeline.<init>(VulkanPipeline.kt:91)
	at graphics.scenery.backends.vulkan.VulkanPipeline.addShaderStages(VulkanPipeline.kt:94)
	at graphics.scenery.backends.vulkan.VulkanRenderer$debugCallback$1.<init>(VulkanRenderer.kt:151)
	at graphics.scenery.backends.vulkan.VulkanRenderer$PresentHelpers.<init>(VulkanRenderer.kt:50)
	at graphics.scenery.backends.vulkan.VulkanRenderer$PresentHelpers.<init>(VulkanRenderer.kt:51)
	at graphics.scenery.backends.vulkan.VulkanRenderer$PresentHelpers.<init>(VulkanRenderer.kt:52)
	at graphics.scenery.backends.vulkan.VulkanRenderer$PresentHelpers.<init>(VulkanRenderer.kt:53)
	at graphics.scenery.backends.vulkan.VulkanRenderer$updateDefaultUBOs$1.accept(VulkanRenderer.kt:2348)
	at graphics.scenery.backends.vulkan.VulkanRenderer$updateDefaultUBOs$1.accept(VulkanRenderer.kt:45)
	at graphics.scenery.backends.vulkan.VulkanRenderer$windowSizeCallback$1.<init>(VulkanRenderer.kt:358)
	at graphics.scenery.backends.vulkan.VulkanRenderer.<init>(VulkanRenderer.kt:207)
	at graphics.scenery.backends.vulkan.VulkanRenderer.<init>(VulkanRenderer.kt:219)
	at graphics.scenery.backends.vulkan.VulkanRenderer.<init>(VulkanRenderer.kt:244)
	at graphics.scenery.backends.vulkan.VulkanRenderer.<init>(VulkanRenderer.kt:245)
	at graphics.scenery.backends.vulkan.VulkanRenderer.<init>(VulkanRenderer.kt:270)
	at graphics.scenery.backends.vulkan.VulkanRenderer.<init>(VulkanRenderer.kt:380)
	at graphics.scenery.backends.vulkan.VulkanRenderer.<init>(VulkanRenderer.kt:383)
	at graphics.scenery.backends.vulkan.VulkanRenderer.createDeviceAndGetGraphicsQueueFamily(VulkanRenderer.kt:1449)
	at graphics.scenery.backends.vulkan.VulkanRenderer.createDeviceAndGetGraphicsQueueFamily(VulkanRenderer.kt:1485)
	at graphics.scenery.backends.vulkan.VulkanRenderer.createVertexBuffers(VulkanRenderer.kt:1777)
	at graphics.scenery.backends.vulkan.VulkanRenderer.createVertexBuffers(VulkanRenderer.kt:1834)
	at graphics.scenery.backends.vulkan.VulkanRenderer.getPhysicalDevice(VulkanRenderer.kt:1412)
	at graphics.scenery.backends.vulkan.VulkanRenderer.initializeNode(VulkanRenderer.kt:544)
	at graphics.scenery.backends.vulkan.VulkanRenderer.prepareRenderpassesFromConfig(VulkanRenderer.kt:1009)
	at graphics.scenery.backends.vulkan.VulkanRenderer.prepareRenderpassesFromConfig(VulkanRenderer.kt:1020)
	at graphics.scenery.backends.vulkan.VulkanRenderer.prepareRenderpassesFromConfig(VulkanRenderer.kt:972)
	at graphics.scenery.backends.vulkan.VulkanRenderer.prepareStandardVertexDescriptors(VulkanRenderer.kt:788)
	at graphics.scenery.backends.vulkan.VulkanRenderer.prepareStandardVertexDescriptors(VulkanRenderer.kt:799)
	at graphics.scenery.backends.vulkan.VulkanRenderer.prepareStandardVertexDescriptors(VulkanRenderer.kt:816)
	at graphics.scenery.backends.vulkan.VulkanRenderer.prepareStandardVertexDescriptors(VulkanRenderer.kt:835)
	at graphics.scenery.backends.vulkan.VulkanRenderer.prepareStandardVertexDescriptors(VulkanRenderer.kt:843)
	at graphics.scenery.backends.vulkan.VulkanRenderer.recordPostprocessRenderCommands(VulkanRenderer.kt:2263)
	at graphics.scenery.backends.vulkan.VulkanRenderer.render(VulkanRenderer.kt:1266)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$commandBufferBacking$1.invoke(VulkanRenderpass.kt:50)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$commandBufferCount$1.invoke(VulkanRenderpass.kt:63)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:68)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:69)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:70)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:71)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:72)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:73)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:74)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:75)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:76)
	at graphics.scenery.backends.vulkan.VulkanRenderpass$VulkanMetadata.<init>(VulkanRenderpass.kt:77)
	at graphics.scenery.backends.vulkan.VulkanRenderpass.<init>(VulkanRenderpass.kt:79)
	at graphics.scenery.backends.vulkan.VulkanRenderpass.initializeDefaultPipeline(VulkanRenderpass.kt:181)
	at graphics.scenery.backends.vulkan.VulkanRenderpass.initializePipeline(VulkanRenderpass.kt:187)
	at graphics.scenery.backends.vulkan.VulkanRenderpass.initializePipeline(VulkanRenderpass.kt:198)
	at graphics.scenery.backends.vulkan.VulkanShaderModule.<init>(VulkanShaderModule.kt:69)
	at graphics.scenery.backends.vulkan.VulkanShaderModule.<init>(VulkanShaderModule.kt:80)
	at graphics.scenery.backends.vulkan.VulkanShaderModule.<init>(VulkanShaderModule.kt:83)
	at graphics.scenery.backends.vulkan.VulkanTexture$VulkanImage.copyFrom(VulkanTexture.kt:79)
	at graphics.scenery.backends.vulkan.VulkanTexture$VulkanImage.copyFrom(VulkanTexture.kt:85)
	at graphics.scenery.backends.vulkan.VulkanTexture$VulkanImage.copyFrom(VulkanTexture.kt:87)
	at graphics.scenery.backends.vulkan.VulkanTexture$VulkanImage.copyFrom(VulkanTexture.kt:88)
	at graphics.scenery.backends.vulkan.VulkanTexture$VulkanImage.copyFrom(VulkanTexture.kt:89)
	at graphics.scenery.backends.vulkan.VulkanTexture$VulkanImage.transitionLayout(VulkanTexture.kt:46)
	at graphics.scenery.backends.vulkan.VulkanTexture$VulkanImage.transitionLayout(VulkanTexture.kt:53)
	at graphics.scenery.backends.vulkan.VulkanTexture.createImage(VulkanTexture.kt:110)
	at graphics.scenery.backends.vulkan.VulkanTexture.createImageView(VulkanTexture.kt:212)
	at graphics.scenery.backends.vulkan.VulkanTexture.createImageView(VulkanTexture.kt:221)
	at graphics.scenery.backends.vulkan.VulkanTexture.createSampler(VulkanTexture.kt:231)
	at graphics.scenery.controls.GLFWMouseAndKeyHandler$cursorCallback$1.<init>(GLFWMouseAndKeyHandler.kt:34)
	at graphics.scenery.controls.GLFWMouseAndKeyHandler$keyCallback$1.<init>(GLFWMouseAndKeyHandler.kt:47)
	at graphics.scenery.controls.GLFWMouseAndKeyHandler$mouseCallback$1.<init>(GLFWMouseAndKeyHandler.kt:75)
	at graphics.scenery.controls.GLFWMouseAndKeyHandler$scrollCallback$1.<init>(GLFWMouseAndKeyHandler.kt:99)
```

Some of these are however related to #42, as the debug allocator simply tracks allocs/deallocs within the same scope.

**Comment by skalarproduktraum** [comment]
Remaining reported leaks are:

```
	at graphics.scenery.backends.vulkan.VU$VU.setImageLayout(VU.kt:206)
	at graphics.scenery.backends.vulkan.VU$VU.setImageLayout(VU.kt:264)
	at graphics.scenery.backends.vulkan.VulkanBuffer.<init>(VulkanBuffer.kt:16)
	at graphics.scenery.backends.vulkan.VulkanBuffer.<init>(VulkanBuffer.kt:9)
	at graphics.scenery.backends.vulkan.VulkanRenderer$debugCallback$1.<init>(VulkanRenderer.kt:164)
	at graphics.scenery.backends.vulkan.VulkanRenderer$updateDefaultUBOs$1.accept(VulkanRenderer.kt:2389)
	at graphics.scenery.backends.vulkan.VulkanRenderer$updateDefaultUBOs$1.accept(VulkanRenderer.kt:45)
	at graphics.scenery.backends.vulkan.VulkanRenderpass.initializeDefaultPipeline(VulkanRenderpass.kt:199)
	at graphics.scenery.backends.vulkan.VulkanRenderpass.initializePipeline(VulkanRenderpass.kt:216)
	at graphics.scenery.backends.vulkan.VulkanShaderModule.<init>(VulkanShaderModule.kt:82)
	at graphics.scenery.backends.vulkan.VulkanShaderModule.<init>(VulkanShaderModule.kt:85)
	at graphics.scenery.controls.GLFWMouseAndKeyHandler$cursorCallback$1.<init>(GLFWMouseAndKeyHandler.kt:34)
	at graphics.scenery.controls.GLFWMouseAndKeyHandler$keyCallback$1.<init>(GLFWMouseAndKeyHandler.kt:47)
	at graphics.scenery.controls.GLFWMouseAndKeyHandler$mouseCallback$1.<init>(GLFWMouseAndKeyHandler.kt:75)
	at graphics.scenery.controls.GLFWMouseAndKeyHandler$scrollCallback$1.<init>(GLFWMouseAndKeyHandler.kt:99)
```

These seem to be wrongly reported, however.

**Comment by skalarproduktraum** [comment]
Resolved with 0e158e85f75dc65c487b74ccbdce597c831a9cf6.
